### PR TITLE
[FLINK-27297][python] Add get_execution_environment method with configuration argument.

### PR DESCRIPTION
**Pull request by deadwind4**
## What is the purpose of the change

Add the StreamExecutionEnvironment#getExecutionEnvironment(Configuration) method in PyFlink

## Brief change log

  - *Add a configuration argument in `get_execution_environment` method*

## Verifying this change

This change added tests and can be verified as follows:

*(example:)*
  - *Added test that validates that get_execution_environment pass configuration is effective*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (PyDocs)

### CODE REVIEW

File: flink-python/pyflink/datastream/stream_execution_environment.py [hunk 1]
```diff
@@ -807,17 +807,30 @@ def register_cached_file(self, file_path: str, name: str, executable: bool = Fal
...
         :return: The execution environment of the context in which the program is executed.
         """
         gateway = get_gateway()
-        j_stream_exection_environment = gateway.jvm.org.apache.flink.streaming.api.environment\
-            .StreamExecutionEnvironment.getExecutionEnvironment()
+        JStreamExecutionEnvironment = gateway.jvm.org.apache.flink.streaming.api.environment \
+            .StreamExecutionEnvironment
+
+        if configuration:
+            j_stream_exection_environment = JStreamExecutionEnvironment.getExecutionEnvironment()
```
[the remark this answers is not in the document]
Comment (deadwind4) [reply flink-python/pyflink/datastream/stream_execution_environment.py]: I find the only one [test_table_environment_api.py](https://github.com/apache/flink/pull/19685/files#diff-51b6ef54d67d0bdb98ffaac4535b82acdce2c0e591164a7018350aeda663ecb3) can be updated

File: flink-python/pyflink/table/tests/test_table_environment_api.py [hunk 2]
```diff
@@ -237,12 +237,12 @@ def setUp(self) -> None:
         from pyflink.datastream import StreamExecutionEnvironment
 
         super(DataStreamConversionTestCases, self).setUp()
-        self.env = StreamExecutionEnvironment.get_execution_environment()
+        config = Configuration()
+        config.setString("akka.ask.timeout", "20 s")
```
Comment (dianfu) [inline flink-python/pyflink/table/tests/test_table_environment_api.py:241]:
```suggestion
        config.set_string("akka.ask.timeout", "20 s")
```

**Review by dianfu — COMMENTED** [review]
@deadwind4 Thanks for the update. LGTM overall. There is a small issue which causes the tests failed. Please take a look~

File: flink-python/pyflink/table/tests/test_table_environment_api.py (hunk 2 above)
Comment (deadwind4) [reply flink-python/pyflink/table/tests/test_table_environment_api.py]: sorry

**Pull request closed without merging**